deps/rabbit/src/rabbit_exchange_type_local_random.erl

```diff
@@ -58,11 +58,12 @@ info(_X) -> [].
 info(_X, _) -> [].
 serialise_events() -> false.
 validate(_X) ->
-    case rabbit_feature_flags:is_enabled(?MODULE) of
+    case rabbit_feature_flags:is_enabled(?MODULE) andalso
+         rabbit_misc:get_env(rabbit, enable_local_random_exchange, true) of
         true ->
             ok;
         false ->
-            rabbit_misc:amqp_error(
+            rabbit_misc:protocol_error(
               precondition_failed,
               "x-local-random exchange feature not available", [],
```

Comment:
It would be great to make this error more specific when the feature is disabled and not just "not available" (due to a disabled feature flag).